src/Modules/LiveRankingModule/Services/LiveRankingService.cs

```diff
@@ -1,259 +1,80 @@
-        await manialinkManager.SendPersistentManialinkAsync("LiveRankingModule.LiveRanking",
-            await GetWidgetDataAsync());
-    }
-
-    public async Task SendManialinkAsync()
-    {
-        await manialinkManager.SendPersistentManialinkAsync("LiveRankingModule.LiveRanking",
-            await GetWidgetDataAsync());
-    }
-
-    public async Task OnEndRoundAsync(RoundEventArgs args)
-    {
-        logger.LogTrace("Round {ArgsCount} ends - RoundsMode: {IsRoundsMode}", args.Count, _isRoundsMode);
-        var liveRanking = await _liveRankingStore.GetFullLiveRankingAsync();
-        var nbFinished = liveRanking.FindAll(x => x.IsFinish);
-        if (nbFinished.Count > 0)
-        {
-            logger.LogTrace("MatchInfo Rounds before: {ArgsCount}", _liveRankingStore.GetMatchInfo().NumRound);
-            _liveRankingStore.IncreaseRoundCounter();
-            logger.LogTrace("MatchInfo Rounds after: {ArgsCount}", _liveRankingStore.GetMatchInfo().NumRound);
-        }
-    }
-
-    public async Task OnBeginMatchAsync()
+    public async Task DetectModeAndRequestScoreAsync()
     {
-        await SendManialinkAsync();
+        _isPointsBased = await matchSettingsService.GetCurrentModeAsync() is not DefaultModeScriptName.TimeAttack;
+        await RequestScoresAsync();
     }
 
-    public async Task OnEndMatchAsync(EndMatchGbxEventArgs args)
-    {
-        await HideManialinkAsync();
-    }
-
-    public async Task OnPodiumStartAsync(PodiumEventArgs args)
-    {
-        await CheckAndSetRoundsMode();
-        await HideManialinkAsync();
-    }
+    public Task RequestScoresAsync()
+        => server.Remote.TriggerModeScriptEventArrayAsync("Trackmania.GetScores");
 
-    public async Task HideNadeoScoreboardAsync()
+    public async Task MapScoresAndSendWidgetAsync(ScoresEventArgs scores)
     {
-        var hudSettings = new List<string>
-        {
-            @"{
-    ""uimodules"": [
-        {
-            ""id"": ""Rounds_SmallScoresTable"",
-            ""visible"": false,
-            ""visible_update"": true
-        }
-    ]
-}"
-        };
-
-        await client.Remote.TriggerModeScriptEventArrayAsync("Common.UIModules.SetProperties", hudSettings.ToArray());
+        await manialinkManager.SendPersistentManialinkAsync(WidgetTemplate,
+            new { settings, isPointsBased = _isPointsBased, scores = await MapScoresAsync(scores) });
     }
 
-    private async Task HideManialinkAsync()
+    public Task<IEnumerable<LiveRankingPosition>> MapScoresAsync(ScoresEventArgs scores)
     {
-        await manialinkManager.HideManialinkAsync("LiveRankingModule.LiveRanking");
+        return Task.FromResult(
+            scores.Players.Take(settings.MaxWidgetRows)
+                .Where(score => score != null)
+                .OfType<PlayerScore>()
+                .Where(score => ScoreShouldBeDisplayedAsync(score).Result)
+                .Select(score => PlayerScoreToLiveRankingPositionAsync(score).Result)
```

Comment:
if these methods are just utilities, then don't make them them async, thats unecceessary overhead and syntax.